assets/bower_components/three-js/three-js.html

```diff
@@ -1418,3 +1418,49 @@
     });
   </script>
 </dom-module>
+
+<dom-module id="three-js-raycaster">
+  <script>
+    Polymer({
+      is: 'three-js-raycaster',
+      properties: {
+        signal: {
+          type: Number
+        },
+        event: {
+          type: String
+        }
+      },
+      newRaycaster: function () {
+        this.object = new THREE.Raycaster();
+
+        var renderer = this.fire('three-js-get-renderer').detail.renderer;
+        var scene = Polymer.dom(this).parentNode.objectParent.scene;
+        var camera = Polymer.dom(this).parentNode.objectParent.camera;
+
+        if (this.event) {
+            var that = this;
+            renderer.domElement.addEventListener(this.event, function (event) {
+                var boundingBox = renderer.domElement.getBoundingClientRect();
+
+                var x = 2 * ((event.clientX - boundingBox.left) / renderer.domElement.clientWidth) - 1;
+                var y = 2 * ((boundingBox.top - event.clientY) / renderer.domElement.clientHeight) + 1;
+                that.object.setFromCamera(new THREE.Vector2(x, y), camera);
+
+                var intersects = that.object.intersectObjects(scene.children);
+                if (intersects.length > 0) {
+                    Escher.send({command: 'signal-update', data: {signalId: that.signal, value: intersects[0].point}});
```

Comment:
Using `Escher.send` directly this way is not how it's done everywhere in Escher.

How it would work in Escher is, you'd fire `sugnal-update` _event_ instead. Then `subscribe` will automatically set up the signal from that element's `signal-update` events... Instead of the way you do it in the `raycaster` function. So instead of `raycaster(signal, options)` you'd have `subscribe(signal, raycaster(options))`... You can also set up a way to "interpret" the data coming in from the browser like [this](https://github.com/shashi/Escher.jl/blob/4dd34f36dd846fe9cc7abafda8d712541b7fd45c/src/basics/behavior.jl#L85-L118)

I admit this could be simpler though - but I need to work on that. I'm okay with getting this in if it serves your purpose, provided it does not stop this package from working in Jupyter.